db_migrate/versions/1dd7cd2978aa_deduplicate_files_with_same_content.py

```diff
@@ -0,0 +1,52 @@
+"""deduplicate files with same content
+
+Revision ID: 1dd7cd2978aa
+Revises: 21a4ca1179da
+Create Date: 2017-07-26 13:54:21.400054
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '1dd7cd2978aa'
+down_revision = '21a4ca1179da'
+branch_labels = None
+depends_on = None
+
+from alembic import op
+import sqlalchemy as sa
+from hashlib import sha256
+import sys
+import zlib
+
+
+def upgrade():
+    content = "dummy content, backward incompatible update"
+    hasher = sha256()
+    hasher.update(content)
+    content_hash = hasher.hexdigest()
+    content = zlib.compress(content,
+                            zlib.Z_BEST_COMPRESSION)
+    content_table = op.create_table('file_contents',
+                                    sa.Column('content_hash', sa.String(),
+                                              primary_key=True),
+                                    sa.Column('content', sa.Binary(),
+                                              nullable=False))
+    op.bulk_insert(content_table,
```

Comment:
This `bulk_insert` should check if there are a file in the database or not. For freshly created databases, the migration script automatically insert this dummy dead weight into the database, which is not good.